opexebo/analysis/gridscore.py

```diff
@@ -148,124 +147,191 @@ def gridscore(aCorr, **kwargs):
+        'spacings' : np.array
+            Spacing of three adjacent fields closest to center in autocorr (in [bins])
+        'spacing' : float
+            Nanmean of 'spacings' in [bins]
+        'orientations' : np.array
+            Orientation of three adjacent fields closest to center in autocorr (in [degrees])
+        'orientations_std' : float
+            Standard deviation of orientations % 60
+        'orientation' : float
+            Orientation of grid in [degrees] (mean of fields of 3 main axes)
+        'positions' : np.array
+            [y,x] coordinates of six fields closest to center
+        'ellipse' : np.array
+            Ellipse fit returning [x coordinate, y coordinate, major radius, minor radius, theta]
+        'ellipse_aspect_ratio' : float
+            Ellipse aspect ratio (major radius / minor radius)
+        'ellipse_theta' : float
+            Ellipse theta (corrected according to previous BNT standard) in [degrees]
+    '''
 
-def _grid_score_stats(bestCorr, **kwargs):
     # Get kwargs
     debug = kwargs.get('debug', False)
     min_orientation = kwargs.get('min_orientation', default.min_orientation)
+    if debug: print('Min orientation: {} degrees'.format(min_orientation))
     min_orientation = np.radians(min_orientation)
-    
+
     # Initialise default output in case stats are uncalculable
-    gs_ellipse_theta = np.nan
-    gs_ellipse = np.nan
-    gs_orientation = np.array([np.nan, np.nan, np.nan])
-    gs_spacing = np.array([np.nan, np.nan, np.nan])
-    
-    
-    # Identify local maxima    
-    regionalMax = morphology.local_maxima(bestCorr, connectivity=4, indices=False)
-    selem = morphology.square(3)
-    dilated_img = morphology.dilation(regionalMax, selem)
-    labelled_img = morphology.label(dilated_img)
-    #labelled_img[labelled_img==int(np.max(labelled_img)/2)+1] = 0
-    
-    if np.max(labelled_img) >= 7:
-        properties = measure.regionprops(labelled_img, cache=True)
-        
-        # Typical shape is 71x71 -> central bin is 35,35
-        centre = -0.5 + np.array(bestCorr.shape)/2 # centre : also [y, x]
-        all_coords = np.array([region.centroid for region in properties])
-        # x-coords are all_coords[:,1], y are [:,0].
-        
-        if debug:
-            plt.figure()
-            plt.title("Labelled autocorrelogram and ellipse")
-            plt.imshow(labelled_img)
-            plt.scatter(all_coords[:,1], all_coords[:,0])
-        
-        
-        # Bearing from field to centre
+    gs_ellipse_theta, gs_ellipse, gs_aspect_ratio = np.nan, np.nan, np.nan
+    gs_orientations_std, gs_orientation = np.nan, np.nan
+    gs_positions = np.array([np.nan, np.nan, np.nan, np.nan, np.nan, np.nan])
+    gs_orientations = np.array([np.nan, np.nan, np.nan])
+    gs_spacings = np.array([np.nan, np.nan, np.nan])
+
+    # Find fields in autocorrelogram
+    acorr_fields = sep.extract(aCorr, mask=mask, thresh=.1) # CONSIDER MAKING THRESHOLD AN ARGUMENT
```

Comment:
I need to test this against the ephys data before making it the mandatory method - previously, it couldn't cope, and I still haven't figured out why

Reply:
Thanks, yes, I would be grateful if you could compare with ephys data. Would be great if we could use the same method here.